### PR TITLE
[TS migration] Migrate 'HeaderGap' component to TypeScript

### PR DESCRIPTION
**Pull request by SzymczakJ**
### Details
Migrate 'HeaderGap' component to TypeScript. During this task i noticed that webpack doesn't include .desktop.tsx files in bundle when running desktop app and desktop app wouldn't work if we wanted to write some desktop specific code, I've put out a [PR that fixes this](https://github.com/Expensify/App/pull/32548). Without that PR all components that have .desktop.tsx files won't work properly.
I also changed SplashScreenHider so that it's typed better(so it doesn't say it returns null, but ReactNode), because it was a similar case to HeaderGap component(more details in comments below).

### Fixed Issues
$ https://github.com/Expensify/App/issues/24990
PROPOSAL: N/A

### Tests

- [X] Test HeaderGap usage in main flows of the app(for example check if HeaderGap exists between Header Back Button and top edge of window(only on desktop) on Profile Page)
- [X] Verify that no errors appear in the JS console

### Offline tests

- N/A

### QA Steps

Same as tests

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/88395093/916eb302-efed-403e-9e9b-4a7969978391

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/88395093/3d017483-f8c1-4c0f-9542-5f89510597c7

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/88395093/aa98a9cb-0e89-494d-a7c3-0c3de29fa4fc

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/88395093/a2d2515e-ced3-4b14-8683-5bc79282cbdc

</details>

### CODE REVIEW

**Review by blazejkustra — APPROVED** [review]

File: src/components/HeaderGap/index.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {ReactNode} from 'react';
+import HeaderGapProps from './types';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function HeaderGap({styles}: HeaderGapProps): ReactNode {
```
Comment (VickyStash) [inline src/components/HeaderGap/index.tsx:5]: We usually don't add return type to components

Comment (blazejkustra) [reply src/components/HeaderGap/index.tsx:5]: Without it the return type would be `null`, that's why it was initially typed like this:
```ts
const HeaderGap: HeaderGapComponent = () => null;
```

Comment (blazejkustra) [reply src/components/HeaderGap/index.tsx:5]: @VickyStash I think we need to adjust [SplashScreenHider](https://github.com/Expensify/App/blob/main/src/components/SplashScreenHider/index.tsx) return type. Default implementation `index.tsx` returns `null`, but native implementation `index.native.tsx` returns an `Element`. The problem is that typescript infers the type from default implementation so whenever you use this component it's treated like it returns `null` which isn't entirely true..
image

> And the component seems to work as expected.

It does, but the type is misleading 😅 

> But if you think it's better to keep your initial typing I'm okay with that!

I think the most clean is to make it an exception and leave the return type for `HeaderGap` and `SplashScreenHider`, as it is now

Comment (blazejkustra) [reply src/components/HeaderGap/index.tsx:5]: @SzymczakJ Let's move the return type to `types.ts` and reuse this type in implementations. Go ahead and adjust [SplashScreenHider](https://github.com/Expensify/App/blob/main/src/components/SplashScreenHider/index.tsx) as well, thanks!

**Review by VickyStash — APPROVED** [review]
LGTM

File: src/components/HeaderGap/index.tsx (hunk 1 above)
Comment (SzymczakJ) [reply src/components/HeaderGap/index.tsx]: Fixed comments! @VickyStash @blazejkustra what do you think?

**Review by Ollyws — APPROVED** [review]
LGTM.

**Review by marcochavezf — APPROVED** [review]
Also LGTM, thank you guys! 🚀

**Pull request merged**